Content.Server/Antag/AntagSelectionSystem.cs

```diff
@@ -341,12 +341,17 @@ public void MakeAntag(Entity<AntagSelectionComponent> ent, ICommonSession? sessi
 
         if (session != null)
         {
-            var curMind = _mind.CreateMind(session.UserId, Name(antagEnt.Value));
-            _mind.SetUserId(curMind, session.UserId);
+            var curMind = session.GetMind(); 
 
-            _mind.TransferTo(curMind, antagEnt, ghostCheckOverride: true);
-            _role.MindAddRoles(curMind, def.MindComponents, null, true);
-            ent.Comp.SelectedMinds.Add((curMind, Name(player)));
+            if (curMind == null || HasComp<GhostComponent>(session.AttachedEntity))
+            {
+                curMind = _mind.CreateMind(session.UserId, Name(antagEnt.Value));
+                _mind.SetUserId(curMind.Value, session.UserId);
+            }
```

Comment:
this should probably be checking if the new antag entity is different from the player's original entity in addition to the null check.

that would cover basically all the times an antagonist changes your entity and requires a new mind to be made.

Reply:
Changed it.
I originally intended to do this but didn't for whatever reason.